dapr/actor/runtime/context.py

```diff
@@ -6,17 +6,28 @@
 """
 
 from dapr.actor.id import ActorId
+from dapr.actor.runtime.stateprovider import StateProvider
+from dapr.clients import DaprActorClientBase, DaprActorHttpClient
 from dapr.serializers import Serializer
 
+from typing import Callable
+
 class ActorRuntimeContext:
     """A context of ActorRuntime"""
 
     def __init__(
             self, actor_type_info: 'ActorTypeInformation',
-            message_serializer: Serializer, actor_factory=None):
+            message_serializer: Serializer, state_serializer: Serializer,
+            actor_client: DaprActorClientBase = None,
+            actor_factory: Callable[['ActorRuntimeContext', ActorId], 'Actor'] = None):
         self._actor_type_info = actor_type_info
-        self._message_serializer = message_serializer
         self._actor_factory = actor_factory or self._default_actor_factory
+        self._message_serializer = message_serializer
+        self._state_serializer = state_serializer
+
+        # Create State management provider for actor.
+        self._dapr_client = actor_client or DaprActorHttpClient()
```

Comment:
You are making this specific to HTTP. Actor API can be in GRPC in the future. Can we make actor_client mandatory here?

Reply:
sure